File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())

board = []
for _ in range(n):
  board.append(list(input()))

def countRepaint(x, y):
  blackStarts = 0
  whiteStarts = 0
  for i in range(8):
    for j in range(8):
      if (i + j) % 2 == 0:
        if board[x + i][y + j] == 'B':
          whiteStarts += 1
        else:
          blackStarts += 1
      else:
        if board[x + i][y + j] == 'W':
          whiteStarts += 1
        else:
          blackStarts += 1
  return min(blackStarts, whiteStarts)

answer = 100000

for i in range(n - 7):
  for j in range(m - 7):
    answer = min(answer, countRepaint(i, j))

print(answer)

File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int n, m;
vector<string> board;

int countRepaints(int x, int y) {
    int blackStarts = 0, whiteStarts = 0;
    for (int i = 0; i < 8; ++i) {
        for (int j = 0; j < 8; ++j) {
            if ((i + j) % 2 == 0) {
                if (board[x + i][y + j] != 'B') blackStarts++;
                if (board[x + i][y + j] != 'W') whiteStarts++;
            }
            else {
                if (board[x + i][y + j] != 'W') blackStarts++;
                if (board[x + i][y + j] != 'B') whiteStarts++;
            }
        }
    }
    return min(blackStarts, whiteStarts);
}

int main() {
    cin >> n >> m;
    board.resize(n);

    for (int i = 0; i < n; ++i) {
        cin >> board[i];
    }

    int answer = 1e9;

    for (int i = 0; i + 7 < n; ++i) {
        for (int j = 0; j + 7 < m; ++j) {
            answer = min(answer, countRepaints(i, j));
        }
    }

    cout << answer << endl;

    return 0;
}